microsoft/testsuites/stress/stress_ng_suite.py

```diff
@@ -112,34 +134,220 @@ def _run_stress_ng_job(
+
+                node_outputs.append(node_output)
+
+            except Exception as e:
+                failed_nodes += 1
+                error_output = f"=== {node_name} ===\nERROR: {str(e)}"
+                node_outputs.append(error_output)
+                log.error(f"{node_name} failed: {e}")
+                # Store the exception to re-raise after collecting all outputs
+                exceptions_to_raise.append(e)
+
+        # Combine all node outputs, including node names for clarity
+        execution_summary = f"Job: {job_file_name}\n\n"
+        for i, node_output in enumerate(node_outputs):
+            node_name = nodes[i].name if i < len(nodes) else f"node-{i + 1}"
+            execution_summary += f"=== {node_name} ===\n{node_output}\n\n"
+
+        # If any processes failed, re-raise the first exception to fail the test
+        if exceptions_to_raise:
+            log.error(
+                f"Stress-ng job failed on {failed_nodes} node(s). "
+                f"Re-raising first exception to fail the test case."
+            )
+            raise exceptions_to_raise[0]
+
+        # Return status and stress-ng info output
+        overall_status = TestStatus.PASSED if failed_nodes == 0 else TestStatus.FAILED
+        return overall_status, execution_summary
+
+    def _report_test_results(
+        self,
+        test_result: TestResult,
+        job_file_name: str,
+        execution_status: TestStatus,
+        execution_summary: str,
+    ) -> None:
+        """
+        Report the stress test results through LISA's messaging system.
+
+        Args:
+            test_result: Test result object for reporting
+            job_file_name: Name of the executed job file
+            execution_status: Final test status (PASSED/FAILED)
+            execution_summary: Comprehensive execution summary
+        """
+        send_sub_test_result_message(
+            test_result=test_result,
+            test_case_name=job_file_name,
+            test_status=execution_status,
+            test_message=execution_summary,
+        )
 
     def _check_panic(self, nodes: List[RemoteNode]) -> None:
         for node in nodes:
             node.features[SerialConsole].check_panic(saved_path=None, force_run=True)
+
+    def _process_yaml_output(
+        self,
+        node: RemoteNode,
+        job_file_name: str,
+        log: Logger,
+    ) -> str:
+        """
+        Process YAML output file if it exists and return a concise summary string.
+        Only extracts 'system-info' and 'times' sections if present.
+        """
+        logging.getLogger("YamlManager").setLevel(logging.WARNING)
+
+        job_stem = Path(job_file_name).stem
+        yaml_filename = f"{job_stem}.yaml"
+        yaml_file_path = node.working_path / yaml_filename
+
+        if not node.shell.exists(yaml_file_path):
+            return "No YAML output file found"
+
+        result = node.execute(f"cat '{yaml_file_path}'", shell=True)
```

Comment:
I believe there is a `Cat` tool you can use here.

Reply:
fixed